test/output-app/model/Tweet.js

```diff
@@ -1,60 +1,115 @@
+import log from '../server/logger';
 import DataLoader from 'dataloader';
-import findByIds from 'mongo-find-by-ids';
+import { findByIds, queryForRoles, authlog, checkAuthDoc } from '../server/authorize';
 
 export default class Tweet {
   constructor(context) {
     this.context = context;
     this.collection = context.db.collection('tweet');
     this.pubsub = context.pubsub;
-    this.loader = new DataLoader(ids => findByIds(this.collection, ids));
+    let authQuery;
+    try {
+      const { me, User } = context;
+      authQuery = queryForRoles(me, ['admin', 'world'], ['authorId', 'coauthorsIds'], { User }, authlog('tweet findOneById', 'readOne', me));
+    } catch (err) { 
+      log.error(err.message);
+      authQuery = {_id: false}; // otherwise admin access
+    }
+    this.authorizedLoader = new DataLoader(ids => findByIds(this.collection, ids, authQuery));
   }
 
-  findOneById(id) {
-    return this.loader.load(id);
+  async findOneById(id, me, resolver) {
+    try {
+      return await this.authorizedLoader.load(id);
+    } catch (err) { log.error(err.message); }
   }
 
-  all({ lastCreatedAt = 0, limit = 10 }) {
-    return this.collection.find({
-      createdAt: { $gt: lastCreatedAt },
-    }).sort({ createdAt: 1 }).limit(limit).toArray();
+  find({ lastCreatedAt = 0, limit = 10, baseQuery = { createdAt: { $gt: lastCreatedAt } } }, me, resolver) {
```

Comment:
This is OK for now, but I think we should probably move toward something a little less ad-hoc in the future: https://github.com/tmeasday/create-graphql-server/issues/41